lib/code_corps/github/api/api.ex

```diff
@@ -2,19 +2,37 @@ defmodule CodeCorps.GitHub.API do
   alias CodeCorps.{
     GithubAppInstallation,
     GitHub,
+    GitHub.API.Pagination,
     GitHub.APIError,
     GitHub.HTTPClientError,
     User
   }
 
+  alias HTTPoison.{Error, Response}
+
+  def gateway(), do: Application.get_env(:code_corps, :github)
+
   @spec request(GitHub.method, String.t, GitHub.headers, GitHub.body, list) :: GitHub.response
-  def request(method, url, headers, body, options) do
-    method
-    |> :hackney.request(url, headers, body, options)
+  def request(method, url, body, headers, options) do
+    gateway().request(method, url, body, headers, options)
     |> marshall_response()
   end
 
-  defdelegate get_all(url, headers, opts), to: CodeCorps.GitHub.EagerAPI
+  @spec get_all(String.t, GitHub.headers, list) :: GitHub.response
+  def get_all(url, headers, options) do
+    {:ok, %Response{request_url: request_url, headers: response_headers}} =
+      gateway().request(:head, url, "", headers, options)
+
+    response_headers
+    |> Pagination.retrieve_total_pages()
+    |> Pagination.to_page_numbers()
+    |> Enum.map(&Pagination.add_page_param(options, &1))
+    |> Enum.map(&gateway().request(:get, url, "", headers, &1))
+    |> Enum.map(&marshall_response/1)
+    |> Enum.map(&Tuple.to_list/1)
+    |> Enum.map(&List.last/1)
+    |> List.flatten
```

Comment:
I just read (oddly in `benchee`'s README) that `Enum.map |> List.flatten` is 1.83x slower than `Enum.flat_map`. Wonder if we should use that instead here?